### PR TITLE
OCPBUGS-32873: Replace deprecated gopkg.in/square/go-jose.v2

### PR DESCRIPTION
**Pull request by sayan-biswas**
Replace deprecated gopkg.in/square/go-jose.v2 with github.com/go-jose/go-jose/v3

Fixes the following CVE(s):
* CVE-2024-28180

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/blang/semver/v4 v4.0.0
 	github.com/containers/image/v5 v5.25.0
 	github.com/davecgh/go-spew v1.1.1
+	github.com/go-jose/go-jose/v3 v3.0.3
```
Comment (avinal) [inline go.mod:12]: Is this okay to use v3 for replacement of v2? I know it is a bump, but just in case if there are breaking changes across releases.

**Review by avinal — COMMENTED** [review]
Rest looks okay to me. One question

**Comment by avinal** [comment]
If you have to cherry-pick, then I suggest you command the bot before PR merges, if seems the bot doesn't work on closed PR.

**Comment by sayan-biswas** [comment]
> If you have to cherry-pick, then I suggest you command the bot before PR merges, if seems the bot doesn't work on closed PR.

But I guess cherry-pick bot is configured to work with minimum "merged" status. Isn't it?

File: go.mod (hunk 1 above)
Comment (sayan-biswas) [reply go.mod]: Done.

**Comment by avinal** [comment]
> But I guess cherry-pick bot is configured to work with minimum "merged" status. Isn't it?

Yes, it will only open cherrypick PR once merged, but the bot ignores command after merging it seems.

**Review by avinal — APPROVED** [review]
LGTM

**Comment by sayan-biswas** [comment]
/cherrypick release-4.16 release-4.15 release-4.14 release-4.13 release-4.12

**Pull request merged**